packages/synapse-react-client/src/components/ImageCardGridWithLinks/ImageCardGridWithLinks.tsx

```diff
@@ -0,0 +1,216 @@
+import { getFiles } from '../../synapse-client/SynapseClient'
+import { parseEntityIdFromSqlStatement } from '../../utils/functions/SqlFunctions'
+import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos'
+
+export type ImageCardGridWithLinksProps = {
+  sql: string
+  title: string
+  summaryText: string
+}
+
+type ImageCardProps = {
+  card: Row
+  index: number
+}
+
+enum ExpectedColumns {
+  LINKTEXT = 'LinkText',
+  LINK = 'Link',
+  IMAGE = 'Image',
+}
+
+function ImageCardGridWithLinks(props: ImageCardGridWithLinksProps) {
+  const { sql, title, summaryText } = props
+  const { accessToken } = useSynapseContext()
+  const [images, setImages] = useState<string[] | undefined>()
+
+  const entityId = parseEntityIdFromSqlStatement(sql)
+  const queryBundleRequest: QueryBundleRequest = {
+    partMask:
+      SynapseConstants.BUNDLE_MASK_QUERY_SELECT_COLUMNS |
+      SynapseConstants.BUNDLE_MASK_QUERY_RESULTS,
+    concreteType: 'org.sagebionetworks.repo.model.table.QueryBundleRequest',
+    entityId,
+    query: {
+      sql,
+    },
+  }
+
+  const { data: queryResultBundle, isLoading } =
+    useGetQueryResultBundle(queryBundleRequest)
+
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        const imageColumnIndex = getFieldIndex(
+          ExpectedColumns.IMAGE,
+          queryResultBundle,
+        )
+        const image = (queryResultBundle?.queryResult!.queryResults.rows.map(
+          el => el.values[imageColumnIndex],
+        ) ?? []) as string[]
+        const imageFileHandleIds = image.filter(
+          v => v != null && v !== undefined,
+        )
+        if (imageFileHandleIds.length === 0) {
+          return
+        }
+        const fileHandleAssociationList: FileHandleAssociation[] =
+          imageFileHandleIds.map(fileId => {
+            return {
+              associateObjectId: entityId,
+              associateObjectType: FileHandleAssociateType.TableEntity,
+              fileHandleId: fileId,
+            }
+          })
+        const batchFileRequest: BatchFileRequest = {
+          includeFileHandles: false,
+          includePreSignedURLs: true,
+          includePreviewPreSignedURLs: false,
+          requestedFiles: fileHandleAssociationList,
+        }
+        const files = await getFiles(batchFileRequest, accessToken)
+        setImages(
+          files.requestedFiles
+            .filter(el => el.preSignedURL !== undefined)
+            .map(el => el.preSignedURL!),
+        )
+      } catch (e) {
+        console.error('Error on get data', e)
+      }
+    }
+    getData()
+  }, [entityId, accessToken, queryResultBundle])
+
+  const dataRows = queryResultBundle?.queryResult!.queryResults.rows ?? []
+  const linkColumnIndex = getFieldIndex(ExpectedColumns.LINK, queryResultBundle)
+  const linkTextColumnIndex = getFieldIndex(
+    ExpectedColumns.LINKTEXT,
+    queryResultBundle,
+  )
+
+  const ImageCard = ({ card, index }: ImageCardProps) => (
```

Comment:
Do not define function components within other function components. This violates [react/no-unstable-nested-components](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unstable-nested-components.md)